vovagusev_code/task_3/face_mask_detection.py

```diff
@@ -0,0 +1,553 @@
+
+import os
+import cv2
+import numpy as np 
+import pandas as pd 
+import matplotlib.pyplot as plt
+import tensorflow as tf
+from tensorflow.keras import layers
+from google.colab import files
+from zipfile import ZipFile
+from tensorflow.keras.optimizers import Adam
+from tensorflow.keras.applications import VGG16
+from tensorflow.keras.applications import VGG19
+from tensorflow.keras.applications import InceptionV3
+from tensorflow.keras.applications import ResNet50
+from tensorflow.keras.applications import InceptionResNetV2
+
+"""**Loading the data**"""
+
+!pip install kaggle
+
+files.upload()
+
+! mkdir ~/.kaggle
+ ! cp kaggle.json ~/.kaggle/
+! chmod 600 ~/.kaggle/kaggle.json
+
+!kaggle datasets download -d omkargurav/face-mask-dataset
+
+file_name = 'face-mask-dataset.zip'
+
+with ZipFile(file_name, 'r') as zip:
+  zip.extractall()
+  print("Done")
+
+batch_size = 40
+img_height = 200
+img_width = 200
+
+train_data = tf.keras.preprocessing.image_dataset_from_directory(
+    'data',
+    validation_split=0.2,
+    subset="training",
+    seed=42,
+    image_size=(img_height, img_width),
+    batch_size=batch_size
+)
+
+test_data = tf.keras.preprocessing.image_dataset_from_directory(
+    'data',
+    validation_split=0.2,
+    subset="validation",
+    seed=42,
+    image_size=(img_height, img_width),
+    batch_size=batch_size
+)
+
+class_names = train_data.class_names
+
+plt.figure(figsize=(10, 10))
+for images, labels in train_data.take(1):
+  for i in range(12):
+    ax = plt.subplot(3, 4, i + 1)
+    plt.imshow(images[i].numpy().astype("uint8"))
+    plt.title(class_names[labels[i]])
+    plt.grid(True)
+
+"""**Configuring datasets for performance**"""
+
+AUTOTUNE = tf.data.experimental.AUTOTUNE
+train_data = train_data.cache().prefetch(buffer_size=AUTOTUNE)
+test_data= test_data.cache().prefetch(buffer_size=AUTOTUNE)
+
+"""**Modeling**
+
+Using flatten layer will increase the size of the model,
+
+hence use Global avg Pooling instead.
+
+Is it necessary to include input_shape in the first layer?
+
+Does it affect the performance?
+"""
+
+model = tf.keras.models.Sequential([
+  layers.experimental.preprocessing.Rescaling(1./255),
+  layers.Conv2D(32, 3, activation='relu'),#,input_shape=(200,200,3)),
+  layers.MaxPooling2D(),
+  layers.Conv2D(64, 3, activation='relu'),
+  layers.MaxPooling2D(),
+  layers.Conv2D(128, 3, activation='relu'),
+  layers.MaxPooling2D(),
+  layers.GlobalAveragePooling2D(),
+  layers.Dense(256, activation='relu'),
+  layers.Dense(2, activation= 'softmax')
+])
+
+model.compile(optimizer='adam', loss='sparse_categorical_crossentropy', metrics=['accuracy'])
+
+"""**Training my CNN**"""
+
+retVal = model.fit(train_data, validation_data=test_data, epochs=15)
+
+"""**Let's visualize the result of training.**"""
+
+plt.plot(retVal.history['loss'], label = 'training loss')
+plt.plot(retVal.history['accuracy'], label = 'training accuracy')
+plt.legend()
+
+"""Is it bad when the learning curve looks almost like a line?
```

Comment:
You compare loss and accuracy at the same plot, that's why it looks a little weird.
For better visualization you could use separate plots for loss and accuracy:
plt.plot(retVal.history['accuracy'], label='train accuracy')
plt.plot(retVal.history['val_accuracy'], label='val accuracy')
plt.legend()